dimod/discrete/discrete_quadratic_model.py

```diff
@@ -810,3 +808,372 @@ def to_numpy_vectors(self, return_offset: bool = False):
+
+        """
+        case = self._lookup_shared_case(v, case)
+        super().set_linear_case(v, case, bias)
+
+    def set_quadratic(self, u, v, biases):
+        """Set biases associated with the interaction between `u` and `v`.
+
+        Args:
+            u: A variable in the discrete quadratic model, or a unique case
+                label.  If `u` is a unique case label, `v` must be a unique
+                case label.
+
+            v: A variable in the discrete quadratic model, or a unique case
+                label.  If `u` is a unique case label, `v` must be a unique
+                case label.
+
+            biases (float or array-like/dict):
+                The quadratic biases.  If `u` and `v` are variables, then
+                `biases` may be a dictionary of the form
+                `{case_u, case_v: bias, ...}` or a
+                :meth:`~DiscreteQuadraticModel.num_cases(u)` by
+                :meth:`~DiscreteQuadraticModel.num_cases(v)` array-like.
+
+                If `u` and `v` are unique case labels, the quadratic bias is a
+                real number.
+
+        Raises:
+            ValueError: If `u` is a unique case label and `v` is not.
+
+        """
+        u_k = self._unique_label_case.get(u)
+        if u_k:
+            if v not in self._unique_label_case:
+                raise ValueError(f'unknown case label {v}')
+
+            v_m = self._unique_label_case[v]
+            super().set_quadratic_case(*u_k, *v_m, biases)
+        else:
+            super().set_quadratic(u, v, biases)
+
+    def set_quadratic_case(self, u, u_case, v, v_case, bias):
+        """Set the bias associated with the interaction between two cases of
+        variables `u` and `v`.
+
+        Args:
+            u: A variable in the discrete quadratic model.
+
+            u_case: The case of `u`.
+
+            v: A variable in the discrete quadratic model.
+
+            v_case: The case of `v`.
+
+            bias (float): The quadratic bias.
+
+        """
+        u_case = self._lookup_shared_case(u, u_case)
+        v_case = self._lookup_shared_case(v, v_case)
+        super().set_quadratic_case(u, u_case, v, v_case, bias)
+
+    def get_cases(self, v):
+        """The cases of variable `v`.
+
+        Returns:
+            List of case labels for `v`, if case labels exist for `v`.
+
+            If case labels do not exist for `v`, returns a list of integers
+            from `0` to :meth:`~DiscreteQuadraticModel.num_cases(v)` - 1.
+
+        """
+        range_ = range(self.num_cases(v))
+        map_ = self._shared_case_label.get(v)
+        if map_:
+            return [map_[case] for case in range_]
+
+        elif v in self._unique_label_vars:
+            return [self._unique_case_label[(v, case)] for case in range_]
+
+        else:
+            return list(range_)
+
+    def to_file(self, *args, **kwargs):
+        raise NotImplementedError
+
+    @classmethod
+    def from_file(cls, *args, **kwargs):
```

Comment:
I think we could actually leave this as-is, and support the
```
dqm = DQM()
...
cldqm = CaseLabelDQM.from_file(dqm.to_file())
```

On the other hand, it might be confusing to support `from_` but not `to_`. Up to you :shrug: